nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/store/strategy/TraceStorageStrategy.java

```diff
@@ -49,14 +49,13 @@
 public interface TraceStorageStrategy {
 
     /**
-     * Get the trace that needs removing.
-     *
-     * @param traces the list of traces.
-     * @param maxSize the maximum size of the list.
-     * @return the trace that should be removed, or null if no traces should be
-     * removed.
+     * Get the trace that needs removing suggesting a candidate that should be chosen in case it is contained in the
+     * given list of traces.
+     * 
+     * @param traces        the list of traces to test.
+     * @param maxSize       the maximum size of the list.
+     * @param traceToRemove the trace to remove if present or null if no particular candidate if favoured
+     * @return the trace that needs removing, or null if no traces need removing.
      */
-    RequestTrace getTraceForRemoval(Collection<RequestTrace> traces, int maxSize);
```

Comment:
As both implementations would just call the other `getTraceForRemoval` with `null` for the `traceToRemove` I felt it could very well be removed and called with `null` directly.